akka-docs/src/test/scala/docs/stream/operators/sourceorflow/WatchTermination.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package docs.stream.operators.sourceorflow
+
+import akka.actor.ActorSystem
+import akka.stream.scaladsl.Source
+
+import scala.concurrent.ExecutionContext
+import scala.util.{ Failure, Success }
+
+object WatchTermination {
+
+  def watchTerminationExample(): Unit = {
+    implicit val system: ActorSystem = ???
+    implicit val ec: ExecutionContext = ???
+
+    //#watchTermination
+    Source(1 to 5)
+      .watchTermination()(
+        (prevMatValue, future) =>
+          // this function will be run when the stream terminates
+          // the Future provided as a second parameter indicates whether the stream completed successfully or failed
+          future.onComplete {
+            case Failure(exception) => println(exception.getMessage)
+            case Success(_)         => println(s"The stream materialized $prevMatValue")
+          })
+      .runForeach(println)
+    /*
+    Prints:
+    1
+    2
+    3
+    4
+    5
+    The stream materialized NotUsed
```

Comment:
It might be nice to add a log message before the 'onComplete' that logs the materialized value, and then just 'stream completed successfully' here - that'd illustrate that the materialized value is created immediately and the future completes when the stream completes.

Also perhaps it'd be nicer to log with 'log.info' rather than 'println'.

Reply:
I think we do println in most of these samples so probably ok IMO

Reply:
Thanks for the advice @raboof! I could maybe have time to add the log message you suggested in the upcoming days. I'll open antoher PR for this.